### PR TITLE
Refresh authentication token when delegate keys change.

### PR DESCRIPTION
**Pull request by iancmcc**
This also modifies the auth code to allow the functions that block waiting for a token or keys to be cancellable.

### CODE REVIEW

File: auth/token.go [hunk 1]
```diff
@@ -52,24 +52,33 @@ func RefreshToken(f TokenFunc, filename string) (int64, error) {
 		return 0, err
 	}
 	updateToken(token, time.Unix(expires, 0), filename)
-	log.WithField("expiration", expires).Debug("Received new authentication token")
+	log.WithField("expiration", expires).Info("Received new authentication token")
```
Comment (johnlhamilton) [inline auth/token.go]: If we leave this as Info, it is probably going to show up as output on every CLI command.

File: makefile [hunk 2]
```diff
@@ -160,7 +160,7 @@ FORCE:
 serviced: $(GO)
 serviced: FORCE
 	$(GO) build $(GOBUILD_FLAGS) ${LDFLAGS}
-	make govet
+	# make govet
```
Comment (jafcrocker) [inline makefile]: We need to run govet

File: auth/token.go (hunk 1 above)
Comment (iancmcc) [inline auth/token.go]: We need CLI commands not to show logging at all.

Comment (johnlhamilton) [inline auth/token.go]: Opened CC-2719 to address this.

**Pull request merged**